readme.md

```diff
@@ -31,8 +31,19 @@ cd build
 cmake ..
 cmake build .
 ```
-
-Now you will be able to build the Wase Engine. When you build the Wase Engine there will be an include and lib folder in the build directory. You can use these for your game projects.
+Now you will be able to build the Wase Engine. 
+
+## Using Wase Engine in your games
+When you build the Wase Engine there will be an include and lib folder in the build directory. You can use these for your game projects.
+
+**MinGW:**
```

Comment:
Why this part? We can just add it to the CMakeLists.txt by default right? Because of the `if(MINGW)` it won't matter right?

Reply:
What I'm talking about is not the game engine's cmake file.  The game's cmake file. `If (MinGW)` matter.

Reply:
To speak a little more clearly, there is no need for any mingw settings in the cmakefile to build the game engine.  However, there is a need for the mingw setting I mentioned in the cmakefile file we use to build the game.

Reply:
I mean this part:
```
if(MINGW)
    target_link_libraries(${PROJECT_NAME} mingw32)
    add_definitions(-DSDL_MAIN_HANDLED)
endif(MINGW)
```
We can just add this by default in the CMakeLists.txt right? If someone uses mingw it uses this code. If they don't use MINGW it skips it. Then people don't have to put it in manually. I guess it works fine when you just put it after the target_link libraries at the end of the file like this:
```
target_link_libraries(wase-engine ${SDL2_LIBRARY}
        ${SDL2_IMAGE_LIBRARIES}
        ${SDL2_MIXER_LIBRARIES}
        ${SDL2_TTF_LIBRARIES})

if(MINGW)
    target_link_libraries(${PROJECT_NAME} mingw32)
    add_definitions(-DSDL_MAIN_HANDLED)
endif(MINGW)

file(COPY src/ DESTINATION ${CMAKE_BINARY_DIR}/include)```

Correct me if I am wrong.

Reply:
Let's say we add the following line of code to the game engine's cmakefile:
```
if(MINGW)
     target_link_libraries(${PROJECT_NAME} mingw32)
endif(MINGW)
#Side Note: I realized, we don't need "add_definitions(-DSDL_MAIN_HANDLED)"
```

Let's say we build the game engine this way. Then let's assume that we have included the lib and include file that has just been created into the game.
Game's cmakefile(pseudocode):
```
cmake_minimum_required(VERSION 3.17)
project(tetris)
set(CMAKE_CXX_STANDARD 20)
find_package(SDL2)
include_directories(SDL2, SDL2image, SDL2ttf, SDL2mixer, wase_engine_include_dir)
add_executable(tetris main.cpp game.h game.cpp)
target_link_libraries(tetris SDL2, SDL2image, SDL2ttf, SDL2mixer, wase_engine_libs)
``` 

Now you think that by entering the command `cmake .. wase_engine_include_dir=path wase_engine_libs=path` we will build the application without any problems. 

But we can't build them. It gives error. 

There were one important things we did when creating the lib file in the first place:
link library: mingw32

The link info is only valid for the game engine's lib file. Not valid for my game files. To build the game, we also need to link mingw32. 

You can try if you want.

Reply:
I still don't get this one. When I add
```
if(MINGW)
     target_link_libraries(${PROJECT_NAME} mingw32)
endif(MINGW)
```
to my CMakeLists.txt I can build with VS and I can build with MinGW. Because if the if statement it just skips it when you don't use MinGW. So we can just add these 3 lines by default to the CMakeLists.txt and remove this part from the readme.md. Where does it give you errors if we do this because it should not give you any.

It works fine for me when I run:
```
cmake ..
cmake build .
and then build in vs
```

and also with:
```
cmake .. -G "MinGW Makefiles"
make
```

So just add the 3 lines by default to the CMakeLists.txt and then we can remove it from the readme.md

Reply:
We have no problem on game engine build. You can add that lines to cmakefile. And you can build with MSVC or whatever compiler.

But lets say you did build game engine with MinGW. We have include folder ve lib folder.

Include folder contains cpp ve h files. lib folder contains `libwase-engine.a`

Now lets try build simon with MinGW.

`cmake -Bbuild -G "MinGW Makefiles* wase_engine_include=path/include wase_engine_libs=path/lib/libwase-engine.a`

This gives error. Because we need change simon's cmakefile too. We need to add that lines again:
```
if(MINGW)
     target_link_libraries(${PROJECT_NAME} mingw32)
endif(MINGW)
```

Reply:
But Simon is an independent project so that is a problem for Simon. If somebody chooses to use mingw to build Wase Engine they need to make sure their CMakeLists.txt is good for their project that uses Wase Engine. So for Wase Engine we can just add it by default into the CMakeLists.txt and remove it from the readme.md

Reply:
Well, its up to you. I am convinced. You can remove changes in readme and change cmakelist. Then you can merge it.